### PR TITLE
Update functions.md

### PR DESCRIPTION
**Pull request by MessageBTilly**
Added instructions for Functions in the dashboard

### CODE REVIEW

File: docs/channels/using_channels/functions.md [hunk 1]
```diff
@@ -78,7 +91,68 @@ exports.handler = async (pusher) => {
...
+
+![Create a new function](../../../img/create-new-function-from-dashboard.png)
+
+5. Finally upload a .zip file of your function code. Click **Create function**. Make sure the event handler is in a local file named index.js, and that all the supporting files are in a directory tree rooted at the same directory as the handler. For example:
+```
+myfunction.zip
+├── index.js
+├── tools.js
+└── utils
+ └── more_tools.js
```
Comment (willpusher) [inline docs/channels/using_channels/functions.md:116]:
```suggestion
    └── more_tools.js
```

File: docs/channels/using_channels/functions.md [hunk 2]
```diff
@@ -78,7 +91,68 @@ exports.handler = async (pusher) => {
...
+2. Choose the Channels app you wish to add functions to.
+3. On the left-side navigation, go to **Functions** and click **Create a new function**.
+
+![Pusher Serverless Functions in the dashboard](../../../img/pusher-serverless-functions-in-dashboard.png)
+
+4. Provide a name for your new function and determine which events should trigger the function. The function needs to know which events to respond to. You provide this as a comma-separated list, where each item is a regular expression. This pattern can match one or multiple event names. Then choose what kind of function it is (synchronous or asynchronous), and finally upload a .zip file of your function code.
+
+![Create a new function](../../../img/create-new-function-from-dashboard.png)
+
+5. Finally upload a .zip file of your function code. Click **Create function**. Make sure the event handler is in a local file named index.js, and that all the supporting files are in a directory tree rooted at the same directory as the handler. For example:
```
Comment (willpusher) [inline docs/channels/using_channels/functions.md:110]:
```suggestion
5. Finally upload a .zip file of your function code. Click **Create function**. Make sure the event handler is in a file named `index.js` at the top level of the zip archive. For example:
```

File: docs/channels/using_channels/functions.md [hunk 3]
```diff
@@ -78,7 +91,68 @@ exports.handler = async (pusher) => {
...
+
+To update a Function, edit the function's code on your computer, then upload it again through the Pusher dashboard.
+
+After you delete a function, it won't be invoked to its linked events in Channels anymore.
+
+![Edit functions](../../../img/create-new-function-from-dashboard.png)
+
+### Configuration variables
+
+To use secret or plaintext configuration variables with you functions, click Add a new config variable.
```
Comment (willpusher) [inline docs/channels/using_channels/functions.md:131]: Does "Add a new config variable" need to be highlighted somehow (e.g. in bold)?

Comment (willpusher) [inline docs/channels/using_channels/functions.md:131]:
```suggestion
To use secret or plaintext configuration variables with your functions, click Add a new config variable.
```

File: docs/channels/using_channels/functions.md [hunk 4]
```diff
@@ -78,7 +91,68 @@ exports.handler = async (pusher) => {
...
+
+Functions can use config variables, either secret or in plain text. You can use it as a secure storage for important data, like third-party API keys. When setting one up, give it a unique name. This name lets you easily use or "call" the key in your functions.
+
+Secret and plaintext configuration variables are available to all the functions in a Channels app.
+
+### Triger functions
```
Comment (willpusher) [inline docs/channels/using_channels/functions.md:137]:
```suggestion
### Trigger functions
```

File: docs/channels/using_channels/functions.md [hunk 5]
```diff
@@ -78,7 +91,68 @@ exports.handler = async (pusher) => {
...
+
+After you delete a function, it won't be invoked to its linked events in Channels anymore.
+
+![Edit functions](../../../img/create-new-function-from-dashboard.png)
+
+### Configuration variables
+
+To use secret or plaintext configuration variables with you functions, click Add a new config variable.
+
+Functions can use config variables, either secret or in plain text. You can use it as a secure storage for important data, like third-party API keys. When setting one up, give it a unique name. This name lets you easily use or "call" the key in your functions.
```
Comment (willpusher) [inline docs/channels/using_channels/functions.md:133]:
```suggestion
Functions can use config variables, either secret or in plain text. You can use secret config variables for sensitive data, like third-party API keys. When setting one up, give it a unique name. This name lets you easily use or "call" the key in your functions.
```

File: docs/channels/using_channels/functions.md [hunk 6]
```diff
@@ -78,7 +91,68 @@ exports.handler = async (pusher) => {
...
+
+Secret and plaintext configuration variables are available to all the functions in a Channels app.
+
+### Triger functions
+
+Define a condition cause for the function to run. Click **Trigger Function** and format this data based on the function's requirements.
```
Comment (willpusher) [inline docs/channels/using_channels/functions.md:139]:
```suggestion
Define a stimulus for the function, formatted based on the function's requirements. Click **Trigger Function** to run the function.
```
This doesn't read comfortably to me, I've suggested an alternative.

File: docs/channels/using_channels/functions.md [hunk 7]
```diff
@@ -78,7 +91,68 @@ exports.handler = async (pusher) => {
...
+
+![Invoke functions](../../../img/invoke-functions-pusher-channels.png)
+
+You can check the function's activity and performance under the **Stats** tab in the Pusher dashboard.
+
+## Functions in Pusher CLI
+
+You can also deploy and manage your functions using the Pusher CLI. Check out the [How to install the Pusher CLI](https://pusher.com/docs/channels/pusher_cli/installation/) and [How to set up the Pusher CLI](https://pusher.com/docs/channels/pusher_cli/documentation/) articles for more information.
+
+Through the Pusher CLI, you can edit, update, delete, and list all functions per app using the --app-id flag or use it to manage secret or plaintext configuration variables.
```
Comment (willpusher) [inline docs/channels/using_channels/functions.md:153]:
```suggestion
Through the Pusher CLI, you can edit, update, delete, and list all functions per app using the `--app-id` flag or use it to manage secret or plaintext configuration variables.
```

File: docs/channels/using_channels/functions.md [hunk 8]
```diff
@@ -21,14 +21,27 @@ There are two ways you can use functions:
     
 -   **Asynchronous** - Functions that operate on events  after the event was published to subscribers. Use cases include integrations with other platforms.
 
-## Manage functions
+## Using functions
+There are two ways you can manage Pusher Serverless Functions:
 
-To deploy Functions on Channel events, you’ll need to use the Pusher CLI. Check out the [How to install the Pusher CLI](/docs/channels/pusher_cli/installation/) and [How to set up the Pusher CLI](/docs/channels/pusher_cli/documentation/) articles for more information.
+- through the the Pusher dashboard,
```
Comment (fbenevides) [inline docs/channels/using_channels/functions.md:27]: Isn't it a good place for a link to Pusher dashboard?

File: docs/channels/using_channels/functions.md [hunk 9]
```diff
@@ -21,14 +21,27 @@ There are two ways you can use functions:
...
 
-Through the [Pusher CLI](/docs/channels/pusher_cli/overview/), you can edit, update, delete, and list all functions per app using the `--app-id` flag or use it to manage secret or plaintext configuration variables.
+- using the Pusher CLI.
```
Comment (fbenevides) [inline docs/channels/using_channels/functions.md:29]: Same for Pusher CLI repo?

File: docs/channels/using_channels/functions.md [hunk 10]
```diff
@@ -78,7 +91,68 @@ exports.handler = async (pusher) => {
...
 
-## Deploy a function
+If you are using the Pusher CLI, skip to [Functions in Pusher CLI](https://pusher.com/docs/channels/using_channels/functions/#functions-in-pusher-cli).
+
+If you are using the Pusher dashboard, when you’re finished writing your function, compress it into a .zip file making sure that the `index.js` file is in the root level, and then upload it via the dashboard following the steps below.
+
+## Functions in the Pusher dashboard
+
+1. To get started with Pusher Channels, [sign in](https://dashboard.pusher.com/accounts/sign_in) to your Pusher account. If you don’t have one, [sign up](https://dashboard.pusher.com/accounts/sign_up) for a free account.
+2. Choose the Channels app you wish to add functions to.
```
Comment (fbenevides) [inline docs/channels/using_channels/functions.md:101]: Since we're explaining how to sign up, is it reasonable to explain how to create a new app?

**Review by ovieokeh — APPROVED** [review]

**Pull request merged**